csv/src/fs2/data/csv/CellEncoder.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fs2.data.csv
+
+import java.net.URL
+import java.time._
+import java.util.UUID
+
+import cats._
+import cats.implicits._
+
+import scala.concurrent.duration.{Duration, FiniteDuration}
+
+/** Describes how a cell of a given type can be encoded.
+  */
+trait CellEncoder[T] {
+  def apply(cell: T): String
+
+  def contramap[B](f: B => T): CellEncoder[B] = (cell: B) => apply(f(cell))
+}
+
+object CellEncoder
+    extends CellEncoderInstances1
+    with LiteralCellEncoders
+    with ExportedCellEncoders {
+
+  implicit object CellEncoderInstances extends Contravariant[CellEncoder] {
+    override def contramap[A, B](fa: CellEncoder[A])(f: B => A): CellEncoder[B] = fa.contramap(f)
+  }
+
+  def apply[T: CellEncoder]: CellEncoder[T] = implicitly[CellEncoder[T]]
+
+  @inline
+  def instance[T](f: T => String): CellEncoder[T] = s => f(s)
+
+  @inline
+  def fromToString[A]: CellEncoder[A] = _.toString
+
+  implicit def enumerationEncoder[E <: Enumeration]: CellEncoder[E#Value] =
+    _.toString
+
+  // Primitives
+  implicit val unitEncoder: CellEncoder[Unit] = _ => ""
+  implicit val booleanEncoder: CellEncoder[Boolean] = _.toString
+  implicit val byteEncoder: CellEncoder[Byte] = _.toString
+  implicit val shortEncoder: CellEncoder[Short] = _.toString
+  implicit val charEncoder: CellEncoder[Char] = _.toString
+  implicit val intEncoder: CellEncoder[Int] = _.toString
+  implicit val longEncoder: CellEncoder[Long] = _.toString
+  implicit val floatEncoder: CellEncoder[Float] = _.toString
+  implicit val doubleEncoder: CellEncoder[Double] = _.toString
+  implicit val bigDecimalEncoder: CellEncoder[BigDecimal] = _.toString
+  implicit val bigIntEncoder: CellEncoder[BigInt] = _.toString
```

Comment:
I wonder whether it is not semantically more readable to use the `fromToString` function you introduced.